notes/connection-lifecycle.md

```diff
@@ -0,0 +1,130 @@
+# Connection lifecycle
+
+## Current implementation
+
+`HTTPConnection` should be instantiated with `host` and `port` of the
+**first origin being connected to** to reach the target origin. This either means
+the target origin itself or the proxy origin if one is desired.
+
+```python
+import urllib3.connection
+
+# Initialize the HTTPSConnection ('https://...')
+conn = urllib3.connection.HTTPSConnection(
+    host="example.com",
+    # Here you can configure other options like
+    # 'ssl_minimum_version', 'ca_certs', etc.
+)
+
+# Set the connect timeout either in the
+# constructor above or via the property.
+conn.timeout = 3.0  # (connect timeout)
+```
+
+If using CONNECT tunneling with the proxy, call `HTTPConnection.set_tunnel()`
+with the tunneled host, port, and headers. This should be called before calling
+`HTTPConnection.connect()` or sending a request.
+
+```python
+conn = urllib3.connection.HTTPConnection(
+    # Remember that the *first* origin we want to connect to should
+    # be configured as 'host' and 'port', *not* the target origin.
+    host="myproxy.net",
+    port=8080,
+    proxy="http://myproxy.net:8080"
+)
+
+conn.set_tunnel("example.com", scheme="http", headers={"Proxy-Header": "value"})
+```
+
+Connect to the first origin by calling the `HTTPConnection.connect()` method.
+If an error occurs here you can check whether the error occurred during the
+connection to the proxy if `HTTPConnection.has_connected_to_proxy` is true.
+If the value is false then the error didn't occur while connecting to a proxy.
+
+```python
+# Explicitly connect to the origin. This isn't
+# required as sending the first request will
+# automatically connect if not done explicitly.
+conn.connect()
+```
+
+After connecting to the origin, the connection can be checked to see if `is_verified` is set to true. If not the `HTTPConnectionPool` would emits a warning. The warning only matters for when verification is disabled, because otherwise an error is raised on unverified TLS handshake.
```

Comment:
```suggestion
After connecting to the origin, the connection can be checked to see if `is_verified` is set to true. If not the `HTTPConnectionPool` would emit a warning. The warning only matters for when verification is disabled, because otherwise an error is raised on unverified TLS handshake.
```